test/e2e/metrics_test.go

```diff
@@ -82,6 +82,40 @@ func (tc *testContext) testPrometheus(t *testing.T) {
 
 }
 
+// testPrometheusEndpointSliceCleanup verifies that no Windows nodes remain in kubelet EndpointSlices
+func (tc *testContext) testPrometheusEndpointSliceCleanup(t *testing.T) {
+	// List all EndpointSlices for the kubelet service in kube-system namespace
+	endpointSlices, err := tc.client.K8s.DiscoveryV1().EndpointSlices("kube-system").List(
+		context.TODO(),
+		metav1.ListOptions{
+			LabelSelector: "kubernetes.io/service-name=kubelet",
+		},
+	)
+	require.NoError(t, err, "error listing EndpointSlices for kubelet service")
+
+	// Verify that no Windows node addresses appear in any EndpointSlice
+	var foundWindowsNodes []string
+	for _, slice := range endpointSlices.Items {
+		for _, endpoint := range slice.Endpoints {
+			// Check if this endpoint references a node
+			if endpoint.TargetRef != nil && endpoint.TargetRef.Kind == "Node" {
+				// Try to get the node to check its OS
+				node, err := tc.client.K8s.CoreV1().Nodes().Get(context.TODO(),
+					endpoint.TargetRef.Name, metav1.GetOptions{})
+				if err == nil {
```

Comment:
shouldn't we catch non-nil errors? Even a node not being found is a problem since it means that the endpointslice isn't synced.